File: ios/Detonator/Sources/Detonator/Renderer/Target.swift
import UIKit

class Target {
    public final let view: UIView
    public var index: Int
    
    init(view: UIView, index: Int) {
        self.view = view
        self.index = index
    }
    
    public func insert(child: UIView) {
        if child == FullScreenModule.view {
            return
        }
        
        let currentChild = index < view.subviews.count
            ? view.subviews[index]
            : nil
        
        if currentChild != nil {
            if child != currentChild {
                view.insertSubview(child, at: index)
            }
        } else {
            view.addSubview(child)
        }
        
        index += 1
    }
    
    public func remove(child: UIView) {
        child.removeFromSuperview()
    }
}

File: android/Detonator/src/main/java/com/iconshot/detonator/renderer/Target.java
package com.iconshot.detonator.renderer;

import android.view.View;
import android.view.ViewGroup;
import android.view.ViewParent;

import com.iconshot.detonator.module.FullScreenModule;

public class Target {
    public final ViewGroup view;
    public int index;

    public Target(ViewGroup view, int index) {
        this.view = view;
        this.index = index;
    }

    public void insert(View child) {
        ViewParent parent = child.getParent();

        if (child == FullScreenModule.view) {
            return;
        }

        View currentChild = view.getChildAt(index);

        if (currentChild != null) {
            if (child != currentChild) {
                if (parent != null) {
                    remove(child);
                }

                view.addView(child, index);
            }
        } else {
            if (parent != null) {
                remove(child);
            }

            view.addView(child);
        }

        index++;
    }

    public void remove(View child) {
        view.removeView(child);
    }
}
